pyriemann/embedding.py

```diff
@@ -0,0 +1,134 @@
+"""Embedding covariance matrices via manifold learning techniques."""
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from pyriemann.utils.distance import distance
+    
+class Embedding(BaseEstimator, TransformerMixin):
+
+    """Embed SPD matrices into an Euclidean space of smaller dimension.
+
+    It uses diffusion maps to embed the SPD matrices into an Euclidean space.
+    The Euclidean distance between points in this new space approximates 
+    the Diffusion distance (also called commute distance) between vertices 
+    of a graph where each SPD matrix is a vertex. 
+
+    Parameters
+    ----------
+    metric : string | dict (default: 'riemann')
+        The type of metric to be used for defining pairwise distance between 
+        covariance matrices. 
+
+    """
+
+    def __init__(self, metric='riemann'):
+        """Init."""
+        self.metric = metric
+
+    def fit(self, X):
+        """Fit.
+
+        Do nothing. For compatibility purpose.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels)
+            ndarray of SPD matrices.
+
+        Returns
+        -------
+        self : Embedding instance
+        The Embedding instance.
+        """
+        return self
+
+    def transform(self, X, eps=None, tdiff=0.0):
```

Comment:
keep things compatible with sklearn API.
eps and tdiff should be set as class arguments